app/services/tunnel.py

```diff
@@ -96,26 +100,35 @@ def over_tunnel_limit(self) -> bool:
 
     def create_tunnel_nomad(self) -> Tuple[str, dict]:
         """Create a tunnel by scheduling an SSH container into the Nomad cluster"""
+        tcp_ports = []
+        for port in self.port_types:
+            if "tcp" in port:
+                tcp_ports.append(self.get_tcp_port())
+            else:
+                tcp_ports.append(0)
+        stripped_ssh_key = "".join(i for i in self.ssh_key if 31 < ord(i) < 127)
+        new_job = render_template(
+            "sshd.j2.json",
+            ssh_key=stripped_ssh_key,
+            box_name=self.subdomain.name,
+            bandwidth=str(self.current_user.limits().bandwidth),
+            base_url=current_app.config["BASE_SERVICE_URL"],
+            port_types=self.port_types,
+            tcp_ports=tcp_ports,
+            tcp_lb_ip=current_app.config["TCP_LB_IP"],
+        )
+        self.nomad_client.jobs.register_job(literal_eval(new_job))
```

Comment:
So `render_template` renders a Python Dict that we then pass through `literal_eval` in order to pass it to register job.   That seems confusing on a couple of levels - is there a way to pass raw json directly to `register_job` - or failing that might we be better off rendering the job spec as JSON and then parsing it and sending it  to register_job?  Even though `literal_eval` is pretty safe I think I'd feel better if we weren't asking the interpreter to do that for us.

Reply:
_I think_  

`return self.request(json=job, method="post").json()`  is the whole body of this method and `self.request` takes a parameter called "data" that you can provide a string too.